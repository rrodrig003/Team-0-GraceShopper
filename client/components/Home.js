import React, { Component } from 'react'
import ProductSlider from './ProductSlider'
import Category from './Category'

const Home = () => {
  return (
    <div>
      HOME!
      <ProductSlider />
      <Category />
    </div>
  )
}

export default Home;
