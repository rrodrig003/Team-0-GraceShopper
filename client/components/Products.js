import React, { Component } from 'react'
import { connect } from 'react-redux'

class Products extends Component {
  
  render() {
    const { products } = this.props
    return (
      <div>
        {
          products.map(product => {
            return (
              <div key={product.id}>{product.name}</div>
            )
          })
        }
      </div>
    )
  }

}

const mapStateToProps = (state) => {
  return {
    products: state.products.products
  }
}

const stateComponent = connect(mapStateToProps)

const connectedProducts = stateComponent(Products);

export default connectedProducts;
