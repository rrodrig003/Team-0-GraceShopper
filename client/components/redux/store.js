import { combineReducers, createStore, applyMiddleware } from 'redux'
import { logger } from 'redux-logger'
import thunkMiddleware from 'redux-thunk';
import axios from 'axios'

// ACTION CONSTANTS
const GET_PRODUCTS = 'GET_PRODUCTS'
const GET_SINGLE_PRODUCT = 'GET_SINGLE_PRODUCT'
const GET_CATEGORIES = 'GET_CATEGORIES'
const GET_SINGLE_CATEGORY = 'GET_SINGLE_CATEGORY'

// ACTION CREATORS
const getProducts = (products) => ({type: GET_PRODUCTS, products})
const getSingleProduct = (id) => ({type: GET_SINGLE_PRODUCT, id})

const getCategories = (categories) => ({type: GET_CATEGORIES, categories})
const getSingleCategory = (id) => ({type: GET_SINGLE_CATEGORY, id})

// STATES
const productState = {
  products: [],
  singleProduct: {}
}

const categoryState = {
  categories: [],
  singleCategory: {}
}

// REDUCERS
const productReducer = (state = productState, action) => {
  switch (action.type) {
    case GET_PRODUCTS:
      return {...state, products: [...state.products, action.products]}
    case GET_SINGLE_PRODUCT:
      // const singleProd = state.products.filter(elem => action.id === elem.id)
      return {...state, singleProduct: action.product}
    default:
      return state;
  }
}

const categoryReducer = (state = categoryState, action) => {
  switch (action.type) {
    case GET_CATEGORIES:
      return {...state, categories: [...state.categories, action.categories]}
    case GET_SINGLE_CATEGORY:
      // const singleCat = state.categories.filter(elem => action.id === elem.id)
      return {...state, singleCategory: action.category}
    default:
      return state;
  }
}

// THUNKS

export const fetchProducts = () => (dispatch) => {
  return axios.get('/api/products')
    .then(res => res.data)
    .then(products => dispatch(getProducts(products)))
    .catch(e => console.error('***ERROR IN fetchProducts:', e))
}

export const fetchCategories = () => (dispatch) => {
  return axios.get('/api/categories')
    .then(res => res.data)
    .then(categories => {
      dispatch(getCategories(categories))
    })
    .catch(e => console.error('***ERROR IN fetchCategories:', e))
}

export const fetchSingleProduct = (id) => (dispatch) => {
  return axios.get(`/api/products/${id}`)
    .then(res => res.data)
    .then(( [product] ) => {
      dispatch(getSingleProduct(product))
    })
    .catch(e => console.error('***ERROR IN fetchSingleProduct:', e))
}

export const fetchSingleCategory = (id) => (dispatch) => {
  return axios.get(`/api/categories/${id}`)
    .then(res => res.data)
    .then(( [category] ) => {
      dispatch(getSingleCategory(category))
    })
    .catch(e => console.error('***ERROR IN fetchSingleCategory:', e))
}

// CREATE STORE
const rootReducer = combineReducers({
  products: productReducer,
  categories: categoryReducer
})

const store = createStore(rootReducer, applyMiddleware(logger, thunkMiddleware))

export default store;
