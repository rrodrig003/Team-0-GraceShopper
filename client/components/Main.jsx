import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Switch, Route, Redirect } from 'react-router';
import { Link } from 'react-router-dom';
import Navbar from '../containers/NavbarContainer';
import Home from './Home';
import Products from '../containers/ProductsContainer';
import Login from '../containers/AuthContainer';
import Cart from '../containers/CartContainer';
import SingleProduct from '../containers/SingleProductContainer';
import AdminCategory from '../containers/AdminCategoryContainer';
import '../stylesheets/index.scss';
import '../stylesheets/header.scss';

const propTypes = {
  onLoad: PropTypes.func.isRequired,
};

export default class Main extends Component {
  componentDidMount() {
    const { onLoad } = this.props;
    onLoad();
  }

  render() {
    return (
      <div className="main-container">
        <header>
          <h1><Link to="/">BAZAAR</Link></h1>
          <Navbar />
        </header>
        <main>
          <Switch>
            <Route exact path="/" component={Home} />
            <Route exact path="/products" component={Products} />
            <Route exact path="/login" component={Login} />
            <Route exact path="/cart" component={Cart} />
            <Route exact path="/admin/categories" component={AdminCategory} />
            <Route path="/products/:productId" component={SingleProduct} />
            <Redirect to="/" />
          </Switch>
        </main>
      </div>
    );
  }
}

Main.propTypes = propTypes;
