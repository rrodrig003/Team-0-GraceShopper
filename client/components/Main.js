import React, { Component } from 'react';
import { Switch, Route } from 'react-router'
import Home from './Home';
import Products from './Products';
import { connect } from 'react-redux';
import { fetchProducts } from './redux/store'

class Main extends Component {

  componentDidMount() {
    this.props.getProducts()
  }

  render() {
    return (
      <div>
        {/* NAVBAR */}
        <main>
          <Switch>
            <Route exact path="/" component={Home} />
            <Route exact path="/products" component={Products} />
          </Switch>
        </main>
      </div>
    )
  }
}

const mapDispatchToProps = dispatch => {
  return {
    getProducts: function() {
      dispatch(fetchProducts())
    }
  }
}

const stateComponent = connect(null, mapDispatchToProps)

const connectedMain = stateComponent(Main)

export default connectedMain;
