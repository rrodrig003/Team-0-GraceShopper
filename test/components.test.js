import React from "react";
import enzyme, { shallow, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { expect } from "chai";
import "jsdom-global/register";

enzyme.configure({ adapter: new Adapter() });

import Main from "../client/components/Main";
import Home from "../client/components/Home";
import Products from "../client/components/Products";

import { Provider } from "react-redux";
import { Router } from "react-router-dom";
import store from "../client/components/redux/store.js";

import { createMemoryHistory } from "history";
const history = createMemoryHistory();

describe("React componnents", () => {
	let component;
	describe("Main", () => {

		it("should render correctly", () => {

      component = mount(
				<Provider store={store}>
					<Router history={history}>
						<Main />
					</Router>
				</Provider>
			);

			expect(component.find("main")).to.have.length(1);
		});
	});

	describe("Home", () => {
		it("should render correctly", () => {
      component = mount(
				<Provider store={store}>
					<Router history={history}>
						<Home />
					</Router>
				</Provider>
			);

			expect(component.find("ProductSlider")).to.have.length(1);
		});
	});

	describe("Products", () => {
		it("should render correctly", () => {
      component = mount(
				<Provider store={store}>
					<Router history={history}>
						<Products />
					</Router>
				</Provider>
			);

			expect(component.find("div")).to.have.length(1);
		});
	});
});
