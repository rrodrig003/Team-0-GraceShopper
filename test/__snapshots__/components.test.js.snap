// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React componnents Main should render correctly with no props 1`] = `
<div>
  <main>
    <div>
      HOME!
      <div>
        PRODUCT SLIDER
      </div>
      <div>
        <div>
          PRODUCT CATEGORY MENU
        </div>
        <ul />
      </div>
    </div>
  </main>
</div>
`;
